CurrencyConverterControl/Data/CurrencyDataProvider.cs

```diff
@@ -38,11 +39,9 @@ public double Convert(string currencyfrom, string currencyto, double amount)
             {
                 if (currencyfrom == currencyto) { return amount; }
                
-               
-                    //httpClient.DefaultRequestHeaders.Accept.Clear();
-                    //httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-
                     string endpoint = String.Format("{0}convert?access_key={1}&from={2}&to={3}&amount={4}", baseURl, access_key, currencyfrom, currencyto, amount);
+                    // fix: Rate limit error when executing requests too quickly 
+                     Thread.Sleep(500);
```

Comment:
It doesn't work for me Has the same -1.

Why don't you just use the currency exchange rates you strore in exchangeRates ? You already have everything you need, and there is no need to make a new API call. This is not a buy/sell rate, so the values can be calculated. You have all rates relative to the USD, so now you can calculate any exchange rate you want. It just a math. 

And this will solve all the issues.